src/main/java/cn/nukkit/command/args/builder/CommandOverloadBuilder.java

```diff
@@ -0,0 +1,50 @@
+package cn.nukkit.command.args.builder;
+
+import cn.nukkit.command.data.CommandParamType;
+import cn.nukkit.command.data.CommandParameter;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.util.LinkedList;
+import java.util.List;
+
+@ToString
+public class CommandOverloadBuilder {
+    @Getter
+    private List<CommandParameter> parameters = new LinkedList<>();
+
+    public CommandOverloadBuilder literal(String name) {
+        parameters.add(new CommandParameter(name, new String[]{name}));
+        return this;
+    }
+
+    public CommandOverloadBuilder literals(String name, String enumName, String[] names) {
```

Comment:
Using varargs here might be cleaner for API consumers? same for the required args method.

Reply:
thats originally how i had it but because i threw the `name` and `enumName` parameters in there, i all looks a bit messy

![image](https://user-images.githubusercontent.com/32024335/81181377-4ac42b00-8fa4-11ea-801e-2b096ac9a4e6.png)